Cargo.toml

```diff
@@ -47,6 +47,7 @@ ctrlc = { version = "3.2.2", features = ["termination"] }
 directories = "4.0.1"
 walkdir = { version = "2", optional = true }
 tempfile = "3.3.0"
+termcolor = "1.1.3"
```

Comment:
I'd prefer if we kept using owo-colors, as we now have two color crates.

owo-colors should work nicely for this. It also has automatic `COLOR` support https://github.com/jam1garner/owo-colors#only-style-on-supported-terminals

Reply:
The only downside to `owo-colors` is I believe it does not support the `always` configuration like `cargo`, and `termcolor` also supports intelligent color support (only print on supported terminals).

Reply:
Ok updated to use `owo-colors`. It's a bit of a mess, but the best I think we can do while still supporting how Cargo does it (allowing you to specify `auto`, `never`, and `always`). So we still have `ColorChoice`:

```rust
/// Whether messages should use color output
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum ColorChoice {
    /// force color output
    Always,
    /// force disable color output
    Never,
    /// intelligently guess whether to use color output
    Auto,
}
```

And then writing the style is as follows:

```rust
// generate the color style
macro_rules! write_style {
    ($stream:ident, $msg_info:expr, $message:expr $(, $style:ident)* $(,)?) => {{
        match $msg_info.color_choice {
            ColorChoice::Always => write!($stream, "{}", $message $(.$style())*),
            ColorChoice::Never => write!($stream, "{}", $message),
            ColorChoice::Auto => write!(
                $stream,
                "{}",
                $message $(.if_supports_color($stream.owo(), |text| text.$style()))*
            ),
        }?;
    }};
}
```

In order to avoid repeating ourselves extensively, we somewhat need wildcard expansion.

Reply:
Update: we might need to switch back to `termcolor`, or use the old logic of `is_atty`. We can't use `if_supports_color` due to the ISC license of `is_ci`, which we need for the `supports-color` feature.